src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadata.php

```diff
@@ -23,21 +23,28 @@ class ArgumentMetadata
     private $isVariadic;
     private $hasDefaultValue;
     private $defaultValue;
+    private $isNullable;
 
     /**
      * @param string $name
      * @param string $type
      * @param bool   $isVariadic
      * @param bool   $hasDefaultValue
      * @param mixed  $defaultValue
+     * @param bool   $isNullable
      */
-    public function __construct($name, $type, $isVariadic, $hasDefaultValue, $defaultValue)
+    public function __construct($name, $type, $isVariadic, $hasDefaultValue, $defaultValue, $isNullable = null)
     {
         $this->name = $name;
         $this->type = $type;
         $this->isVariadic = $isVariadic;
         $this->hasDefaultValue = $hasDefaultValue;
         $this->defaultValue = $defaultValue;
+        $this->isNullable = (bool) $isNullable;
+
+        if (null === $isNullable) {
+            @trigger_error(sprintf('As of 3.2 the $isNullable argument has been added to %s with a default value. As of 4.0 the argument will become mandatory.', __CLASS__), E_USER_DEPRECATED);
```

Comment:
Minor but: I think both `$isNullable` and `%s` should be enclosed with double quotes (like in the `ArgumentResolver` exception you've tweaked  for instance). :)